mc-sema/cfgToLLVM/x86Instrs.cpp

```diff
@@ -169,12 +169,144 @@ InstTransResult liftInstrImpl(InstPtr ip, BasicBlock *&block, NativeBlockPtr nb,
     }
   } else {
     // Instruction translation not defined.
-    errs() << "Unsupported opcode " << opcode << "!";
+    errs() << "Unsupported opcode " << opcode << " ("
+      << IP->getOpcodeName(opcode) << "): ";
     IP->printInst( &inst, errs(), to_string<VA>(ip->get_loc(), hex));
-    // Print out the unhandled opcode.
-    errs() << to_string<VA>(ip->get_loc(), hex) << " ";
-    errs() << strOut.str() << "\n";
-    errs() << inst.getOpcode() << "\n";
+    errs() << "\n";
+
+    // In the case that we can't find the opcode, try building it out with
```

Comment:
I'd like to see a comment here showing what the generated inline assembly looks like. What I am curious about is: can you splat actual disassembly into the inline asm block, then add a fixup prologue/epilogue that marshals the values to/from the register state structure. This *may* be what you're already doing, and if so, a comment in the code would illustrate that.